plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/HudiSnapshotDirectoryLister.java

```diff
@@ -0,0 +1,82 @@
+package io.trino.plugin.hudi.query;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.metastore.Column;
+import io.trino.metastore.HiveMetastore;
+import io.trino.metastore.Table;
+import io.trino.plugin.hive.HiveColumnHandle;
+import io.trino.plugin.hudi.HudiTableHandle;
+import io.trino.plugin.hudi.partition.HiveHudiPartitionInfo;
+import io.trino.plugin.hudi.partition.HudiPartitionInfo;
+import io.trino.plugin.hudi.storage.TrinoStorageConfiguration;
+import org.apache.hudi.common.config.HoodieMetadataConfig;
+import org.apache.hudi.common.engine.HoodieLocalEngineContext;
+import org.apache.hudi.common.model.FileSlice;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.table.view.FileSystemViewManager;
+import org.apache.hudi.common.table.view.HoodieTableFileSystemView;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class HudiSnapshotDirectoryLister implements HudiDirectoryLister {
+    private final HoodieTableFileSystemView fileSystemView;
+    private final List<Column> partitionColumns;
+    private final Map<String, HudiPartitionInfo> allPartitionInfoMap;
+
+    public HudiSnapshotDirectoryLister(
+            HudiTableHandle tableHandle,
+            HoodieTableMetaClient metaClient,
+            boolean enableMetadataTable,
+            HiveMetastore hiveMetastore,
+            Table hiveTable,
+            List<HiveColumnHandle> partitionColumnHandles,
+            List<String> hivePartitionNames,
+            String commitTime)
```

Comment:
can remove `commitTime` as we are already calling `fsView.getLatest...`

Reply:
Good catch.  I kept it due to older code.  Removed now.